src/Spryker/Zed/ProductStorage/Business/Storage/ProductAbstractStorageWriter.php

```diff
@@ -376,13 +381,22 @@ protected function storeProductAbstractStorageEntity(
      */
     protected function isActive(array $productAbstractLocalizedEntity)
     {
-        foreach ($productAbstractLocalizedEntity['SpyProductAbstract']['SpyProducts'] as $productEntity) {
-            if ($productEntity['is_active']) {
-                return true;
+        $idProductAbstract = $productAbstractLocalizedEntity['fk_product_abstract'];
+
+        if (!array_key_exists($idProductAbstract, self::$activeMap)) {
+            $activeConcretesCount = $this->queryContainer
+                ->queryProductConcreteByProductIds([$idProductAbstract])
+                ->filterByIsActive(true)
+                ->count();
+
+            if ($activeConcretesCount > 0) {
```

Comment:
static::$activeMap[$idProductAbstract] = $activeConcretesCount > 0;

Please, make sure that this we are using `static` here and should also use in our places

Reply:
done

Reply:
Can we make it as oneliner as I mentioned in the example?
I mean the `static::$activeMap[$idProductAbstract] = $activeConcretesCount > 0;`

Reply:
missed that, done now